File: elixir/bob/bob.exs
defmodule Teenager do
  def hey(content) do
    cond do
      Message.silence?(content) ->
        "Fine. Be that way!"
      Message.shouting?(content) ->
        "Woah, chill out!"
      Message.question?(content) ->
        "Sure."
      true ->
        "Whatever."
    end
  end
end

defmodule Message do

  def shouting?(content) do
    String.upcase(content) == content
  end

  def question?(content) do
    String.ends_with? content, "?"
  end

  def silence?(content) do
    String.strip(content || "") == ""
  end

end

File: clojure/bob/bob.clj
(ns bob
  (:require [clojure.string :as string]))


(defn- shouting? [content]
  (= (string/upper-case content) content))

(defn- question? [content]
  (= \? (last content)))

(defn- silence? [content]
  (string/blank? content))

(defn response-for [content] 
  (cond 
    (silence? content ) "Fine, be that way."
    (shouting? content) "Woah, chill out!"
    (question? content) "Sure."
    :else "Whatever."))

